storage/pinlogs_blockwise.h

```diff
@@ -0,0 +1,88 @@
+
+
+#if FLASH_BLOCK_WORDS <= 1
+#error "FLASH_BLOCK_WORDS must be at least 2 to fit the counter and header"
+#endif
+
+#if PIN_MAX_TRIES > 16
+#error "PIN_MAX_TRIES must be at most 16"
+#endif
+
+#define PIN_LOG_BYTES ((FLASH_BLOCK_WORDS - 1) * sizeof(uint32_t))
+
+static uint8_t expand_counter(uint8_t c) {
+  c = ((c << 2) | c) & 0x33;
+  c = ((c << 1) | c) & 0x55;
+  c = ((c << 1) | c) ^ 0xaa;
+  return c;
+}
+
+static uint8_t compress_counter(uint8_t c) {
```

Comment:
As I mentioned in a comment in the commit, there should be a format check:
```suggestion
static uint8_t compress_counter(uint8_t c) {
  if ((c ^ (c << 1)) & 0xAA != 0xAA) {
    handle_fault("ill-formed counter");
  }
```

Reply:
added https://github.com/trezor/trezor-firmware/commit/918f6bcdb092d030c7959194fb4107ac0bfde8fb